src/graphql/fanout.ts

```diff
@@ -0,0 +1,307 @@
+  definition (t) {
+    t.nonNull.string('splTokenAddress', {
+      description: 'SPL Token Address',
+    });
+    t.nonNull.string('splTokenWallet', {
+      description: 'SPL Token Wallet',
+    });
+  }
+})
+
+export const CreateFanoutResult = objectType({
+  name: 'CreateFanoutResult',
+  description: 'The result for minting a NFT',
+  definition (t) {
+    t.nonNull.string('message', {
+      description: 'Operation message',
+    });
+    t.string('fanoutPublicKey', {
+      description: 'Fanout public key',
+    });
+    t.field('solanaWalletAddress', {
+      type: 'String',
+      description: 'Solana address of the fanout',
+    });
+    t.field('splFanout', {
+      type: 'SplFanout',
+      description: "Spl Fanout Details"
+    })
+    t.field('splFanoutAddress', {
+      type: 'String',
+      description: 'Spl address of the fanout',
+    });
+    t.field('splWalletAddress', {
+      type: 'String',
+      description: 'Spl address of the fanout',
+    });
+  },
+});
+
+export const DisperseFanoutResult = objectType({
+  name: 'DisperseFanoutResult',
+  description: 'The result for minting a NFT',
+  definition (t) {
+    t.nonNull.string('message', {
+      description: 'Operation message',
+    });
+  },
+});
+
+export const FanoutMember = extendInputType({
+  type: 'FanoutMember',
+  definition (t) {
+    t.nonNull.field('publicKey', {
+      type: 'String',
+      description: 'Public key of member address',
+    });
+    t.nonNull.field('shares', {
+      type: 'Int',
+      description: 'Share member should receive',
+    });
+  },
+});
+
+export const CreateFanout = mutationField('createFanout', {
+  type: 'CreateFanoutResult',
+  args: {
+    encryptedMessage: nonNull(
+      arg({
+        type: 'EncryptedMessage',
+      }),
+    ),
+    name: nonNull(
+      arg({
+        type: 'String',
+      }),
+    ),
+    members: nonNull(
+      arg({
+        type: list('FanoutMember'),
+      }),
+    ),
+    splTokenAddresses: arg({
+      type: list('String'),
+    }),
+  },
+  async resolve (_, args, ctx: YogaInitialContext) {
+    let authorityWallet: Wallet = null!;
+    const connection = new Connection(process.env.RPC_ENDPOINT, 'confirmed');
+    let fanoutSdk: FanoutClient;
+    let splFanoutResult: {splTokenAddress: PublicKey | null, splTokenWallet: PublicKey | null }[] = [];
+
+    const keyPairBytes = JSON.parse(
+      decryptEncodedPayload(args.encryptedMessage),
+    ) as number[];
+
+    // Get create wallet from the client secrets
+    try {
+      authorityWallet = new Wallet(
+        Keypair.fromSecretKey(Uint8Array.from(keyPairBytes)),
+      );
+    } catch (e) {
+      return {
+        message: `Error creating wallet from client secret: ${e.message}`,
+      };
+    }
+
+    // Make sure sum of shares is 100
+    if (
+      args.members.reduce((partialSum, a) => partialSum + a.shares, 0) !== 100
+    ) {
+      return {
+        message: 'Total share must be 100',
```

Comment:
Do we need this restriction? since shares are integers, this limits that number of "shareholders" to 100, I believe the program just uses (basically) member_shares / total_shares to calculate the distribution amount
https://github.com/GlassEaters/hydra/blob/7fc0fa992ed970791072639bf0be079a1eadd68c/programs/hydra/src/utils/logic/calculation.rs#L10

Reply:
You raise a good point, tbh im unsure - i was following whats outlined in the docs and what I saw in `cardinal-labs/hydra-ui`

docs segment: 
```Wallet - This is the simplest membership model. It's just a list of each Member's public address and the number of shares they own. The sum of all Member's shares must equal the totalShares specified in the initializeFanout call.```

Reply:
yeah, I saw that the used a hard coded `100` in the example on hydra docs, but my reading of it was that as long as you pass the sum of all member shares to `initializeFanout` then you need not restrict share divisions to a total of 100.